src/lib.rs

```diff
@@ -35,11 +35,17 @@ impl PID {
     pub fn get(&self) -> u8 {
         self.0
     }
+
+    /// Return if the associated frame uses the classic checksum (diagnostic ID 0x3c-0x3d or
+    /// special use 0x3e, 0x3f)
+    pub fn uses_classic_checksum(&self) -> bool {
+        (self.0 & 0x3c) == 0x3c
```

Comment:
you should use `(self.0 & 0xfc) == 0x3c` since you care about the upper two bits. This will also match `0x7c-0x7f` and `0xfc-0xff`

Reply:
No, the upper bits are parity bits of the id (the lower 6 bits). I must ignore those for the check to work. But maybe it's clearer if I add a `get_id` function that reports the id.